### ISSUE TITLE
`input radio` saves the whole dict in model

### ISSUE THREAD

**Issue by kumy** [kaliber5/ember-bootstrap#769]
I'm trying to use an `input radio` in a form, having label explicitely defined, and expecting a numeric id to be stored in `model`.

I've declared so far:
```hbs
    {{form.element
      id="moveType"
      model=move
      name="moveType"
      controlType="radio"
      label="Move type"
      options=moveTypeOptions
      optionLabelPath="label"
      property="type"
    }}
```
```js
export default Controller.extend({
  moveTypeOptions: [{
      value: 0,
      label: "I've dropped a GeoKret"
    }, {
      value: 1,
      label: "I've grabbed a GeoKret"
    }, {
      value: 3,
      label: "I've met GeoKret"
    }, {
      value: 5,
      label: "I've dipped a GeoKret"
    }, {
      value: 2,
      label: "Comment"
    }]
});
```
Now when an option is selected, the `model.type` is filled with the whole dict: `{value: 2,label: "Comment"}`. It would be better to have only value `2` in model. I would have expected to find a specific option in order to give the path of the value to be stored such as it's done for label (`optionLabelPath`), but didin't found something like that when exploring the code and templates. What about calling it `optionValuePath`? So we could write:
```hbs
    {{form.element
      id="moveType"
      model=move
      name="moveType"
      controlType="radio"
      label="Move type"
      options=moveTypeOptions
      optionLabelPath="label"
      optionValuePath="value"
      property="type"
    }}
```

```
$ ember bootstrap:info
DEPRECATION: ember-cli-babel 5.x has been deprecated. Please upgrade to at least ember-cli-babel 6.6. Version 5.2.8 located: geokrety-frontend -> ember-flag-icon-css -> ember-cli-babel
Npm packages:
ember-bootstrap: ^2.6.0 -> 2.6.0
ember-cli: ^3.8.1 -> 3.8.1
bootstrap: n/a
bootstrap-sass: ^3.4.1 -> 3.4.1
ember-cli-sass: ^10.0.0 -> 10.0.0
ember-cli-less: n/a
Bower packages:
bootstrap: n/a
bootstrap-sass: n/a
ember-bootstrap configuration:
bootstrapVersion: 3
importBootstrapFont: true
importBootstrapCSS: false
```

**Comment by jelhan** [comment]
Did you considered using a computed property? It's not only about setting `type` to the `id` of your object but also to map `type` to an object of your array that is selected.

**Comment by jelhan** [comment]
If you only pass hashes for formatting the value, using `{{el.control}}` as a contextual component would be better IMO. But this seems to be broken currently:

- Due to a typo in template (`hasblock` instead of `hasBlock`) it can't be used in block mode currently.
- It does not yield the option nor the index which makes it quite useless.

**Comment by kumy** [comment]
@jelhan Thanks for fixing the `radio` template (that was another issue I encountered) :+1:.

Regarding your comment about using a computed property to map to the model. I can't understand how it works. And the current doc doesn't mention the use of CP. All examples found at [FormElement](https://www.ember-bootstrap.com/api/classes/Components.FormElement.html) page refer only to some sort of "automatic bound" to the model. This seems partially working as having `property="type"` successfully mapped to the `model.type` field (seen in Ember Inspector) but remaining problem is to select only one property from the passed `options`.

**Comment by jelhan** [comment]
@kumy Please find an Ember Twiddle demonstrating the computed property approach here: https://ember-twiddle.com/c3006689d5e9f512d74a2614a5c424e1?openFiles=controllers.application.js%2C

But I would still recommend a contextual component approach that should be possible after #770:

```hbs
{{#form.element
  controlType="radio"
  label="Move type"
  options=moveTypeOptions
  optionLabelPath="label"
  property="selectedOption"
  as |el|
}}
  {{#el.control as |option|}}
    {{get moveTypeLabels option}}
  {{/el.control}}
{{/form.element}}
```

```js
export default Component.extend({
  moveTypeLabels: computed(function() {
    return {
      0: "I've dropped a GeoKret",
      1: "I've grabbed a GeoKret",
      ...
  })
});
```

**Comment by kumy** [comment]
Thanks for the complete examples. Both are working, and effectively contextual component approach is much simpler.

Just for the records, I had to change to:
```hbs
{{#form.element
  controlType="radio"
  label="Move type"
  options=(array 0 1 3 5 2)
  property="type"
  as |el|
}}
…
```

And sorry for making you waste your time.
